### ISSUE TITLE
Make `DynamoDBCrudPolicy` Include Global Secondary Indexes

### ISSUE THREAD

**Issue by alexanderbanks** [awslabs/serverless-application-model#280]
I'm trying to give a Lambda access to my DynamoDB table, and am running into an error about not having permissions on the GSI. Wouldn't it make more sense to include GSIs on the `DynamoDBCrudPolicy` rather than requiring someone to make a new policy, just to add access to a GSI?

AFAICT it's as easy as changing:
             `"arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${tableName}"`

To:

`"arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${tableName}/*"`

And, moreso, from a semantics standpoint, if a Table has a GSI attached, wouldn't you assume that `DynamoDBCrudPolicy` would include it?

**Comment by brettstack** [comment]
I assume appending `/*` doesn't break any of the existing permissions? Should be a simple PR for anyone willing.

**Comment by clickyotomy** [comment]
Hello! Opened https://github.com/awslabs/serverless-application-model/pull/620 for this, can you please take a look?

**Comment by keetonian** [comment]
The [documentation](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/using-identity-based-policies.html) suggests using
```
arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${tableName}
arn:${AWS::Partition}:dynamodb:${AWS::Region}:${AWS::AccountId}:table/${tableName}/index/*
```
like @mhart commented to access both the table and its indexes.

@clickyotomy can you verify that adding `/*` does indeed still work to access both the table and the indexes? This will also allow access to the streams; does it make sense to and do we want to allow access to those in this policy as well?

**Comment by clickyotomy** [comment]
@keetonian - Yes, you are right. I will update the PR and restrict access to the table and it's indexes only.

I checked if changing `Resource` to `...table/${tableName}/*` would allow access to table indexes, but it does **not**; adding `...table/${tableName}/index/*` works.

**Comment by brysontyrrell** [comment]
I just ran into this. This is painful.

**Comment by keetonian** [comment]
Sorry, shouldn't have removed that label already or closed this issue. This is being released in v1.10.0.

**Comment by keetonian** [comment]
Released!
